scipy/_lib/meson.build

```diff
@@ -178,5 +181,55 @@ py3.install_sources(
   subdir: 'scipy/_lib/array_api_compat/torch',
 )
 
+py3.install_sources(
+  [
+    'cobyqa/cobyqa/__init__.py',
+    'cobyqa/cobyqa/framework.py',
+    'cobyqa/cobyqa/main.py',
+    'cobyqa/cobyqa/models.py',
+    'cobyqa/cobyqa/problem.py',
+    'cobyqa/cobyqa/settings.py',
+  ],
+  subdir: 'scipy/_lib/cobyqa',
+)
+
+py3.install_sources(
+  [
+    'cobyqa/cobyqa/tests/__init__.py',
```

Comment:
I don't think we want to install the tests of the separate `cobyqa` package. We should rely only on the tests of the `scipy.optimize` module, because that's the only public API.

Reply:
Related: `scipy/_lib/cobyqa` should be added to the `norecursedirs` listing in `pytest.ini` in the root of the repo.

Reply:
This is updated in the most recent commit. I remember that before including the tests of COBYQA in `meson.build`, some CI was complaining that some pure Python files (the tests of COBYQA) were not included in `meson.build`. I keep this conversation open until we solve this issue.

Cheer,
Tom.

Reply:
I'll have a look at that now. I think we have to whitelist those test files so it's okay with that checker that they're not installed.